lib/resourcecontrol/resource_control.go

```diff
@@ -0,0 +1,179 @@
+package resourcecontrol
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+)
+
+// NOTE: calling log.Fatal() inside a library is usually considered bad
+// practice, in this case it only happens in the child process so the library
+// user will not be affected.
+
+const (
+	execEnvVar      = "OVERSEER_EXEC"
+	cpuMaxEnvVar    = "OVERSEER_CPU_MAX"
+	memMaxEnvVar    = "OVERSEER_MEM_MAX"
+	ioMaxRbpsEnvVar = "OVERSEER_IO_MAX_RBPS"
+	ioMaxWbpsEnvVar = "OVERSEER_IO_MAX_WBPS"
+
+	controlSubtree = "overseer"
+)
+
+var (
+	errCgroupNotMounted = errors.New("cgroup2 is not mounted")
+)
+
+// TODO: add more limits
+type ResourceLimits struct {
+	CPUMax    string
+	MemMax    string
+	IOMaxRbps string
+	IOMaxWbps string
+}
+
+// init will look for a special environment variable that, if present, will
+// signal the need to set up the resource limits and call unix.Exec() to start
+// the limited command
+func init() {
+	if os.Getenv(execEnvVar) == execEnvVar {
+		// Unset custom environment variables
+		for _, v := range []string{execEnvVar, cpuMaxEnvVar, memMaxEnvVar, ioMaxRbpsEnvVar, ioMaxWbpsEnvVar} {
+			os.Unsetenv(v)
+		}
+
+		setResourceLimits()
```

Comment:
I don't understand why you are unsetting all of these env vars before calling `setResourceLimits()`, which looks like it tries to read them? Am I misunderstanding?

Reply:
You are right, that was a last-minute change and I made a mistake.